active/0012-cluster-call.md

```diff
@@ -0,0 +1,179 @@
+# Cluster call transaction
+
+## Changelog
+
+* 2021-08-11: @zhongwencool Initial draft
+
+## Abstract
+
+When EMQX updates the cluster resources via HTTP API,  it first updates the local node resources, and then updates all other nodes via RPC Multi Call to ensure the consistency of resources (configuration)  in the cluster.
+
+**In order to ensure consistency, it must ensure that the updates will be be eventually applied on all nodes in the cluster**.
+
+## Motivation
+
+The current solution is to update the resources of the local node successfully, and then RPC calls to update the resources of other nodes synchronously.
+
+Update resources may be lost during the RPC call.
+
+- If there is a network disturbance during RPC, it may cause the RPC to fail.
+- If a remote operation to update a resource fails, there is no retry or any other remedy, causing inconsistent configuration in the cluster.
+- If multiple updates are performed concurrently, it may happen that node 1 performs updates in order 1, 2, 3, but node 2 updates in order 1, 3, 2. There is no order guarantee.
+- Lack of replay. If a node is down for a while, there is a lack of history event replay to catch up with the changes happened during the down time.
+
+## Design
+
+This proposal uses mnesia to record the execution status of MFA, to ensure the consistency of the final resources & data in the cluster.
+
+This proposal is not applicable to high frequency request calls, all updates are performed in strict order.
+
+### mnesia table structure
+
+```erlang
+-record(cluster_rpc, {tnx_id :: pos_integer(), mfa :: mfa()}).
```

Comment:
maybe add two more fields for troubleshooting & logging. `timestamp`, and `initiator` (the node).